GuiApp/coreapplication.h

```diff
@@ -39,8 +41,57 @@ class CoreApplication : public QApplication
         if (event->type() == QEvent::FileOpen) {
             QFileOpenEvent *openEvent = static_cast<QFileOpenEvent *>(event);
             mFileToLoad = openEvent->file();
+            mMainWindow->loadFile(openEvent->file());
         }
 
         return QApplication::event(event);
     }
+
+    int exec() {
+        SettingsImpl settings(new DefaultSettingsFactory);
+
+        if (!settings.readSetting(SettingTag::licenseAccepted, false).toBool())
+        {
+            auto dummy = MainWindowFactory::getLicenseDialog();
+            dummy->setModal(true);
+            dummy->setWindowTitle(TITLE_BASE);
+            dummy->exec();
+            if ( dummy->result() == QDialog::Rejected) {
+                return 0;
+            }
+            settings.writeSetting(SettingTag::licenseAccepted, true);
+        }
+
+        if (!settings.readSetting(SettingTag::licenseAccepted, false).toBool())
```

Comment:
This looks like a copy & paste error - the license showing code block seems to be duplicated.